pulser-core/pulser/sequence/_seq_drawer.py

```diff
@@ -44,100 +46,107 @@
 
 @dataclass
 class ChannelDrawContent:
-    """The contents for drawingflake a single channel."""
+    """The contents for drawing a single channel."""
 
-    time: list[int]
-    amplitude: list[float]
-    detuning: list[float]
-    phase: list[float]
+    samples: ChannelSamples
     target: dict[Union[str, tuple[int, int]], Any]
-    measurement: Optional[str] = None
     interp_pts: dict[str, list[list[float]]] = field(default_factory=dict)
 
     def __post_init__(self) -> None:
+        self.samples.phase = self.samples.phase / (2 * np.pi)
+        self._samples_from_curves = {
+            "amplitude": "amp",
+            "detuning": "det",
+            "phase": "phase",
+        }
         self.curves_on = {"amplitude": True, "detuning": False, "phase": False}
 
     @property
     def n_axes_on(self) -> int:
         """The number of axes to draw for this channel."""
         return sum(self.curves_on.values())
 
+    def get_input_curves(self) -> list[np.ndarray]:
+        """The samples for the curves, as programmed."""
+        return self._give_curves_from_samples(self.samples)
+
+    def get_output_curves(self, ch_obj: Channel) -> list[np.ndarray]:
+        """The modulated samples for the curves."""
+        mod_samples = self.samples.modulate(ch_obj)
+        return self._give_curves_from_samples(mod_samples)
+
+    def get_interpolated_curves(
+        self, sampling_rate: float
+    ) -> list[np.ndarray]:
+        """The curves with a fractional sampling rate."""
+        indices = np.linspace(
+            0,
+            self.samples.duration - 1,
+            int(sampling_rate * (self.samples.duration + 1)),
+            dtype=int,
+        )
+        sampled_curves = [curve[indices] for curve in self.get_input_curves()]
+        t = np.arange(self.samples.duration)
+        return [CubicSpline(indices, curve)(t) for curve in sampled_curves]
+
     def curves_on_indices(self) -> list[int]:
         """The indices of the curves to draw."""
         return [i for i, qty in enumerate(CURVES_ORDER) if self.curves_on[qty]]
 
+    def _give_curves_from_samples(
+        self, samples: ChannelSamples
+    ) -> list[np.ndarray]:
+        return [
+            getattr(samples, self._samples_from_curves[qty])
+            for qty in CURVES_ORDER
+        ]
+
```

Comment:
Its there because of the naming conflict, right?

Reply:
Yes, pretty much.